app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -115,54 +115,62 @@ def update_auth_header
       response.headers.merge!(auth_header)
 
     else
-
-      ensure_pristine_resource do
-        # Lock the user record during any auth_header updates to ensure
-        # we don't have write contention from multiple threads
-        @resource.with_lock do
-          # should not append auth header if @resource related token was
-          # cleared by sign out in the meantime
-          return if @used_auth_by_token && @resource.tokens[@client_id].nil?
-
-          # determine batch request status after request processing, in case
-          # another processes has updated it during that processing
-          @is_batch_request = is_batch_request?(@resource, @client_id)
-
-          auth_header = {}
-
-          # extend expiration of batch buffer to account for the duration of
-          # this request
-          if @is_batch_request
-            auth_header = @resource.extend_batch_buffer(@token, @client_id)
-
-            # Do not return token for batch requests to avoid invalidated
-            # tokens returned to the client in case of race conditions.
-            # Use a blank string for the header to still be present and
-            # being passed in a XHR response in case of
-            # 304 Not Modified responses.
-            auth_header[DeviseTokenAuth.headers_names[:"access-token"]] = ' '
-            auth_header[DeviseTokenAuth.headers_names[:"expiry"]] = ' '
-
-          # update Authorization response header with new token
-          else
-            auth_header = @resource.create_new_auth_token(@client_id)
-          end
-
-          # update the response header
-          response.headers.merge!(auth_header)
-
-        end # end lock
-      end # end ensure_pristine_resource
+      unless @resource.valid?
+        puts "INVALID #{@resource.errors.full_messages}"
+        @resource = resource_class.find(@resource.to_param) # errors are keep after reload
+        # if we left the model in a bad state, something is wrong in our app
+        raise DeviseTokenAuth::Errors::InvalidModel, 'cannot set auth token in invalid model' unless @resource.valid?
```

Comment:
Maybe put into exception's message information about errors `@resource.errors`? It will be more clearly for user. Something like this:
```ruby
raise DeviseTokenAuth::Errors::InvalidModel,
"cannot set auth token in invalid model. Errors: #{@resource.errors}" unless @resource.valid?
```

That do you think?

Reply:
the puts above are debugging purposes, the recent changes are breaking the tests